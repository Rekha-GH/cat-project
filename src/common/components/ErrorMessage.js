import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { changeError } from '../../cat/Store/CatBundle';
import { getErrorMessage } from '../../cat/Store/CatSelector';

const DEFAULT_CLASSNAME = 'error-message';
const ErrorMessage = React.memo((props) => {
    const error = useSelector(getErrorMessage);
    const dispatch = useDispatch();
    const time = React.useRef(undefined);

    React.useEffect(() => {
        clearTimeout(time.current);
        time.current = setTimeout(() => {
            dispatch(changeError(undefined));
        }, 5000);
    }, [dispatch, error])

    if (error) {
        return <div className={DEFAULT_CLASSNAME}>{error}</div>;
    }
    else {
        return null;
    }
});

export default ErrorMessage;