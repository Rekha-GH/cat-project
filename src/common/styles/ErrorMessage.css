.error-message {
  padding: 5px;
  border: 1px solid #ef5350;
  background: #e57373;
  border-radius: 5px;
}
