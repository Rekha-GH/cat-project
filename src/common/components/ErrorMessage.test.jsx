import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import * as ReactRedux from "react-redux";
import ErrorMessage from "./ErrorMessage";

Enzyme.configure({ adapter: new Adapter() });

describe("ErrorMessage Test", () => {
  const useSelectorMock = jest.spyOn(ReactRedux, "useSelector");
  jest.spyOn(ReactRedux, "useDispatch");

  test("ErrorMessage Call Correctly when ErrorMessage is not available", () => {
    useSelectorMock.mockReturnValue(undefined);
    const wrapper = shallow(<ErrorMessage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("ErrorMessage Call Correctly when ErrorMessage is available", () => {
    useSelectorMock.mockReturnValueOnce("error message");
    const wrapper = shallow(<ErrorMessage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
