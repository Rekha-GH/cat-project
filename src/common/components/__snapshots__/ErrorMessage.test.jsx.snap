// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Test ErrorMessage Call Correctly when ErrorMessage is available 1`] = `
<div
  className="error-message"
>
  error message
</div>
`;

exports[`ErrorMessage Test ErrorMessage Call Correctly when ErrorMessage is not available 1`] = `""`;
